### ISSUE TITLE
Using dnGrep explorer extension on a folder whose name containing commas results in broken value for 'Search in Folder:'

### ISSUE THREAD

**Issue by benc64** [dnGrep/dnGrep#1096]
Steps to reproduce in v4.1.92.0:

1. Create a folder like:  W:\a,b,c,d
2. Right click on the folder in Windows Explorer and select dnGrep from the context menu
3. Confirm that the Search in Folder looks like "W:\a;b;c;d"
4. If you attempt to do a search you might get the error "The search path in the 'Folder:' fields is not valid"

PS. Love dnGrep!

**Comment by benc64** [comment]
For what it's worth, If I drag and drop the folder from Explorer to dnGrep, the commas in the folder name are not converted to semicolons.

**Comment by benc64** [comment]
The following command line methods also result in a "Search in Folder" that looks like "W:\a;b;c;d"
```
C:\Program Files\dnGrep>dngrep /f "W:\a,b,c,d"
C:\Program Files\dnGrep>dngrep /f W:\a,b,c,d
```

**Comment by doug24** [comment]
OK, I'll have to rewrite the path parser.  It's unfortunate that comma and semi-colon are valid path characters since they are also the most common list separators.

**Comment by benc64** [comment]
some command line parsers I have seen (in other languages) use syntax like the following for repeated parameters:
`dngrep.exe /f "C:\dir a,b" "C:\dir c"`
or alternatively
`dngrep.exe /f "C:\dir a,b" /f "C:\dir c"`

**Comment by doug24** [comment]
I like the second one, I had not thought of doing that.  I think I'll add that to what I've already done. 

I have something working now by checking to see if the string following the comma or semi-colon is a path root. Since the paths have to rooted, it checks to see if they are of one of the following:
`c:` (or any other upper- or lower-case letter)
`\\` which covers a UNC path `\\host\share` or a long path `\\?\C:\`
I think those are the only possible starting sequences for a path root, and they contain a character that cannot be in a file or directory name.

**Comment by doug24** [comment]
Fixed in [v4.2.6](https://github.com/dnGrep/dnGrep/releases/tag/v4.2.6.0)
